packages/flutter_test/lib/src/all_elements.dart

```diff
@@ -23,15 +25,42 @@ Iterable<Element> collectAllElementsFrom(
   return CachingIterable<Element>(_DepthFirstChildIterator(rootElement, skipOffstage));
 }
 
+/// Provides a recursive, efficient, depth first search of an element tree.
+///
+/// [Element.visitChildren] does not guarnatee order, but does guarnatee stable
+/// order. This iterator also guarantees stable order, and iterates in a left
+/// to right order:
+///
+///       1
+///     /   \
+///    2     3
+///   / \   / \
+///  4   5 6   7
+///
+/// Will iterate in order 1, 2, 4, 5, 3, 6, 7. A ListQueue is used to allow for
+/// efficient add/remove First/Last and avoid using List.reversed, which has
+/// slightly worse performance.
+///
+/// Performance is important here because this method is on the critical path
+/// for flutter_driver and package:integration_test performance tests.
+/// Performance is measured in the all_elements_bench microbenchmark.
+/// Any changes to this implementation should check the before and after numbers
+/// on that benchmark to avoid regressions in general performance test overhead.
+///
+/// If we could use RTL order, we could save on performance, but numerous tests
+/// have been written (and developers clearly expect) that LTR order will be
+/// respected.
 class _DepthFirstChildIterator implements Iterator<Element> {
   _DepthFirstChildIterator(Element rootElement, this.skipOffstage)
-    : _stack = _reverseChildrenOf(rootElement, skipOffstage).toList();
+    : _stack = ListQueue<Element>() {
```

Comment:
nit: move this initialization to where the final field is declared? Shouldn't make a difference, but makes the code slightly more readable, I think, since this init doesn't depend on anything in the constructor?

Reply:
Done